webhooks-extension/endpoints/sink.go

```diff
@@ -187,17 +188,23 @@ func createPipelineRunFromWebhookData(buildInformation BuildInformation, r Resou
 
 	imageTag := buildInformation.SHORTID
 	imageName := fmt.Sprintf("%s/%s", registryURL, strings.ToLower(buildInformation.REPONAME))
-	releaseName := fmt.Sprintf("%s-%s", strings.ToLower(buildInformation.REPONAME), buildInformation.SHORTID)
+	releaseName := ""
+
+	if wantedReleaseName != "" {
+		log.Printf("A specified release name %s has been provided", wantedReleaseName)
+		releaseName = wantedReleaseName
+	} else {
+		log.Printf("Generating release name based on repository name and short commit iD")
+		releaseName = fmt.Sprintf("%s-%s", strings.ToLower(buildInformation.REPONAME), buildInformation.SHORTID)
```

Comment:
So the question I would have here is an architectural one ..... if no release name is provided should it be the case that:

a) project name + commit is used so multiple helm installs are attempted (probably service name clashes)
b) project name is used without commit so upgrade works ... i.e default to a fixed name based on repo

Additional note - I would log the generated name, rather then logging one will be generated and not logging it

Reply:
I've added the generated name log, for the architectural bit, yeah I like b). Indeed with a) you're going to have service name clashes. I'll implement that.